pkg/controller/sls/project_controller.go

```diff
@@ -81,41 +73,17 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 		return nil, errors.New(errNotProject)
 	}
 
-	var (
-		sel    *xpv1.SecretKeySelector
-		region string
-	)
-
-	switch {
-	case cr.GetProviderConfigReference() != nil:
-		if err := c.usage.Track(ctx, mg); err != nil {
-			return nil, errors.Wrap(err, errTrackUsage)
-		}
-
-		pc := &v1alpha1.ProviderConfig{}
-		if err := c.client.Get(ctx, types.NamespacedName{Name: cr.Spec.ProviderConfigReference.Name}, pc); err != nil {
-			return nil, errors.Wrap(err, errGetProviderConfig)
-		}
-		if s := pc.Spec.Credentials.Source; s != xpv1.CredentialsSourceSecret {
-			return nil, errors.Errorf(errFmtUnsupportedCredSource, s)
-		}
-		sel = pc.Spec.Credentials.SecretRef
-		region = pc.Spec.Region
-	default:
-		return nil, errors.New(errNoProvider)
-	}
-
-	if sel == nil {
-		return nil, errors.New(errNoConnectionSecret)
+	pcName := cr.Spec.ProviderConfigReference.Name
```

Comment:
Don't we also need to track usage here via `c.usage` before referencing the `ProviderConfig`?

Reply:
@hasheddan's comment of abstracting these common operations together as a function would have proven to be useful here.

Reply:
Thanks @zzxwill for abstracting the common operations like usage tracking, credential fetching etc. in a common function!